zerver/lib/streams.py

```diff
@@ -728,7 +747,7 @@ def public_stream_user_ids(stream: Stream) -> set[int]:
     return set(active_non_guest_user_ids(stream.realm_id)) | guest_subscriptions_ids
```

Comment:
This function likely needs to be revised to consider the `can_add_users` group having guests as members.